### PR TITLE
Support DirectCast and TryCast expressions in VB IMethodXML

### LINKED ISSUE #7002: VB IMethodXML does not support DirectCast(...) or TryCast(...)
If the InitializeComponent() method of a VB WinForms application contains a `DirectCast(...)` or `TryCast(...)`, the designer fails to load. This is due to the fact that we failed to support these cast types in the `IMethodXML` that is used to set up the designer.

image

Originally filed on Microsoft Connect [here](https://connect.microsoft.com/VisualStudio/feedback/details/1843863/the-form-designer-cant-be-opened-due-to-a-backward-compatibility-issue-introduced-in-vs2015).

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes #7002

Tagging @dotnet/roslyn-ide

### CODE REVIEW

File: src/VisualStudio/Core/Impl/CodeModel/MethodXml/AbstractMethodXmlBuilder.cs [hunk 1]
```diff
@@ -226,6 +228,19 @@ private AttributeInfo RankAttribute(int rank)
             return new AttributeInfo(RankAttributeName, rank.ToString());
         }
 
+        private AttributeInfo SpecialCastKindAttribute(SpecialCastKind? specialCastKind = null)
+        {
+            switch (specialCastKind)
+            {
+                case SpecialCastKind.DirectCast:
+                    return new AttributeInfo(DirectCastAttributeName, "yes");
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Impl/CodeModel/MethodXml/AbstractMethodXmlBuilder.cs]: "yes" ?  not "true"?

Comment (DustinCampbell) [inline src/VisualStudio/Core/Impl/CodeModel/MethodXml/AbstractMethodXmlBuilder.cs]: Correct. That's what WinForms expects from IMethodXml, and we use "yes"/"no" for other attributes as well (like "implicit"). FWIW, I created the unit tests by running IMethodXml in VS 2013 and capturing the output.

**Comment by Pilchie** [comment]
:+1:

**Comment by CyrusNajmabadi** [comment]
:+1:

**Pull request merged**